crates/meilisearch/src/search/mod.rs

```diff
@@ -635,7 +635,7 @@ impl SearchQueryWithIndex {
 pub struct SimilarQuery {
     #[deserr(error = DeserrJsonError<InvalidSimilarId>)]
     #[schema(value_type = String)]
-    pub id: ExternalDocumentId,
+    pub id: serde_json::Value,
```

Comment:
I find that a bit strange to store a `Value` in the structure, shouldn’t we use the [`try_from`](https://meilisearch.github.io/deserr/attributes/field.html#deserrtry_from) attribute here?

Something like that;

```suggestion
    #[deserr(try_from(&serde_json::Value) = ExternalDocumentId::try_from -> DeserrJsonError<InvalidSimilarId>, error = DeserrJsonError<InvalidSimilarId>)]
    #[schema(value_type = String)]
    pub id: ExternalDocumentId,
```

Reply:
I could not get it to work. The `try_from` returns a `milli::UserError`, not a `DeserrJsonError<InvalidSimilarId>`.

It used to return an `InvalidSimilarId`, but the same `ExternalDocumentId` is now used by two routes, so we cannot hardcode the error code anymore.

Reply:
Huum ok I’m sad but it makes sense